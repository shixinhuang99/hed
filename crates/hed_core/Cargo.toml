[package]
name = "hed_core"
version = "0.1.0"
edition = "2021"

[lib]
doctest = false

[dependencies]
thiserror = "1.0.63"
