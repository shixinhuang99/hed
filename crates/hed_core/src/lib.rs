pub mod channel;
pub mod error;
mod hed;
mod hosts_info;

pub use hed::Hed;
pub use hosts_info::HostsInfo;
