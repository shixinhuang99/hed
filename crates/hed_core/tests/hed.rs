use std::env;

use hed_core::Hed;
use insta::assert_debug_snapshot;

#[tokio::test]
async fn test_hed_parse() {
	let cwd = env::current_dir().unwrap();
	let hosts_path = cwd.join("fixture/hosts");
	let mut hed = Hed::new(hosts_path);

	hed.parse().await.unwrap();

	assert_debug_snapshot!("ip_hosts_map", hed.ip_hosts_map);
}
