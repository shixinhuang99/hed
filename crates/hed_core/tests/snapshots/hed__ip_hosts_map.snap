---
source: crates/hed_core/tests/hed.rs
expression: hed.ip_hosts_map
---
{
    "172.16.254.1": IpHosts {
        ip: "172.16.254.1",
        hosts: {
            "a.com",
            "c.com",
            "d.com",
        },
    },
    "2402:1200:4f00:1234:0000:5678:9abc:def0": IpHosts {
        ip: "2402:1200:4f00:1234:0000:5678:9abc:def0",
        hosts: {
            "b.com",
        },
    },
    "127.0.0.1": IpHosts {
        ip: "127.0.0.1",
        hosts: {
            "e.com",
        },
    },
}
